server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderConfig.java

```diff
@@ -54,9 +54,6 @@
   @JsonProperty("numBootstrapThreads")
   private Integer numBootstrapThreads = null;
 
-  @JsonProperty("locationSelectorStrategy")
-  private StorageLocationSelectorStrategy locationSelectorStrategy;
```

Comment:
I think the previous implementation is better since you can use the configuration name `druid.segmentCache.locationSelectorStrategy` without `type`. Is there a reason that `locationSelectorStrategy` cannot be here?

Reply:
This is because `StorageLocationStrategy` depends on `SegmentLoaderConfig.locations`. If `StorageLocationStrategy` is placed here, when `SegmentLoaderConfig` is being deserialized, `locations` required by `StorageLocationStrategy` can't be found and injected into it.

After moving this property out of `SegmentLoaderConfig`, both `SegmentLoaderConfig` and `StorageLocationStrategy` are deserialized during construction of `SegmentLoaderLocalCacheManager`, and jackson could find the `locations` objects required by strategy object through google guice framework to create strategy object correctly.

Using the configuration name without `type` I think is wrong. Please take a look at this configuration `druid.coordinator.balancer.strategy`, or the clarification on the change of configuration name I left above.

Reply:
Got it. Then, how about using a similar naming to the coordinator balancer? For example, we can bind `StorageLocationSelectorStrategy` to `druid.segmentCache.locationSelector`, and use a `strategy` property name for `StorageLocationSelectorStrategy` instead of `type`.

Reply:
Sorry for the late reply.  

Your suggestion makes the naming more meaningful. I'll update this PR later this day.